### DISCUSSION TITLE
Struggling to work with the repo

### DISCUSSION THREAD

**Discussion by GeekMeUpScotty** [SerafinTech/ST-node-ethernet-ip#57] [General]
Hi,

I am working on a pull request to tweak the behavior when unknown tags are "encountered".
I did the modifications "live" on an installation, by modifying the javascript files under src.
However I want to create a pull request so I can install the package properly.
I am absolutely n00b to typescript and the complete ecosystem around it. So I am struggling creating the actual package. While I believe my code changes are sane, I want to test before blindly pushing to npm (or what ever the nomenclature is). What is the best approach to achieve this?
Currently I am trying to simply do the compilation on the target directly. I installed typescript and ran:
`tsc -p ./TSConfig.json`

This is failing with:

```
4.269 src/browser/index.ts(64,27): error TS2769: No overload matches this call.
4.269   Overload 1 of 2, '(intervalId: string | number | Timeout): void', gave the following error.
4.269     Argument of type 'Timer' is not assignable to parameter of type 'string | number | Timeout'.
4.269       Property '[Symbol.dispose]' is missing in type 'Timer' but required in type 'Timeout'.
4.269   Overload 2 of 2, '(id: number): void', gave the following error.
4.269     Argument of type 'Timer' is not assignable to parameter of type 'number'.
4.269 src/browser/index.ts(81,23): error TS2769: No overload matches this call.
4.269   Overload 1 of 2, '(intervalId: string | number | Timeout): void', gave the following error.
4.269     Argument of type 'Timer' is not assignable to parameter of type 'string | number | Timeout'.
4.269   Overload 2 of 2, '(id: number): void', gave the following error.
4.269     Argument of type 'Timer' is not assignable to parameter of type 'number'.
4.269 src/controller/index.ts(849,5): error TS2416: Property 'forEach' in type 'Controller' is not assignable to the same property in base type 'ENIP'.
4.269   Types of parameters 'callback' and 'fn' are incompatible.
4.269     Type 'void | Promise<void>' is not assignable to type '{}'.
4.269       Type 'void' is not assignable to type '{}'.
```

None of these errors seems related to my modifications... So how do I proceed? Or rather: how am I supposed to compile this?
Any help appreciated

**Comment by GeekMeUpScotty** [comment]
So I realized that tsc errors does not necessarily prevent tsc from generating code.
I am closing this discussion again! :)

**Comment by SerafinTech** [comment]
I can't reproduce your issue on the current version. What changes did you make?

**Reply by GeekMeUpScotty** [reply]
I closed the discussion... :) While tsc complains about errors - those errors do not prevent compilation. I have submitted a pull request with the changes in question.
